src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
-export interface Figure {
-
+enum ShapeType {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
 
+export interface Figure {
+  shape: ShapeType,
+  color: Color,
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: ShapeType = ShapeType.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side must be more than 0');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('It is not a triangle');
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiPerimeter
+      * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b)
+      * (semiPerimeter - this.c),
+    );
 
+    return Math.floor(area * 100) / 100;
```

Comment:
The rounding method used for the area is not correct. It's using Math.floor which rounds down, but the task requires rounding to the nearest hundredths. It should use the toFixed() method and convert the result back to number.
Apply for similar cases as well